README.rst
.. include:: _resources.rst

.. _hiveeyes-readme:

######
README
######

.. tip::

    You might want to `continue reading on our documentation space <https://hiveeyes.org/docs/system/>`_,
    the rendering will work properly there.


Change documentation
====================
::

    git clone git@github.com:hiveeyes/documentation.git
    cd documentation
    <editor> *.rst

Build local docs
----------------
::

    <package-manager> install python2-virtualenv
    make docs-html
    <browser> doc/build/html/index.html

Build global docs
-----------------
::

    git commit && git push


Write reStructuredText
======================

How to link to BERadio_ and Kotori_ project documentation?::

    :ref:`beradio-spec`
    :ref:`beradio:beradio-spec`
    :ref:`beradio spec <beradio:beradio-spec>`



CHANGES.rst
#######
CHANGES
#######

.. hint::

    For getting the whole picture of development, please also have a look at
    :ref:`Hiveeyes Arduino changelog <arduino:changes>`,
    :ref:`BERadio changelog <beradio:changes>` and
    :ref:`Kotori changelog <kotori:changes>`.


in-progress
===========
- Update “other projects” page


2017-02-06 0.9.0
================
- Improve docs about :ref:`data-acquisition`
- Add page about our :ref:`foss-contributions`
- Announce the community forum on the :ref:`contact` page
- Update the :ref:`other-projects` page. Thanks to everybody who contributed to it!
- Retroactively update the :ref:`news` page
- Update the :ref:`hiveeyes-research` and :ref:`hiveeyes-tasks` pages


2016-09-21 0.8.1
================
- Improve page :ref:`backend-setup`.


2016-09-16 0.8.0
================
- Improve :ref:`handbook`, :ref:`other-projects`, :ref:`goals` and index pages.
- Add :ref:`contact` and preliminary :ref:`daq-python` page.


2016-09-15 0.7.1
================
- Update page :ref:`other-projects`


2016-09-14 0.7.0
================
- Improve contents of index page, remove defunct live-data display
- Add page about :ref:`other-projects`
- Improve menu structure
- Update email address
- This and that


2016-09-13 0.6.1
================
- Update a bit of documentation and history about August/September 2016


.. _version-0.6.0:

2016-07-10 0.6.0
================
- Update a bit of documentation and history about June/July 2016


.. _version-0.5.0:

2016-06-17 0.5.0
================
- Layout, Referencing, Wording, Fixes
- Improve description of the :ref:`projects’ origin <project-origin>`
- Add :doc:`vendor/beutenkarl/index` page
- :ref:`handbook`:

    - Clarify the purpose and communication paths of the two different Open Hive sensor node flavours
    - How to send a :ref:`kotori:sawtooth-signal` from the workbench for testing purposes
    - Reference the two flavours of beehive scales :ref:`beutenkarl-scale` and :ref:`openhive-scale`

- Add page about :ref:`mqttwarn-alerts`
- Add some research information, mainly around LoRa and related technologies
- Some words about software watchdog for :ref:`data-loss events`
- Enable :ref:`Data acquisition over HTTP <daq-http>` and provide an example for :ref:`daq-php`
- Enable automatic firmware building, see :ref:`arduino:firmware-builder`


.. _version-0.4.1:

2016-05-26 0.4.1
================
- Clarify event about nightly weight gain
- Improve :ref:`do-it-yourself`
- Improve landing page :ref:`hiveeyes-system`
- This and that
- Add reference to Arduino documentation
- Reference Arduino documentation via intersphinx
- Disable caching of remote intersphinx inventories
- This and that


2016-05-26 0.4.0
================
- Embed live Grafana dashboard into :ref:`history`
- Update link to Kotori_
- Add link to `Hiveeyes Arduino`_ Atmel AVR MCU code on GitHub
- Improve index page by using information from former splash screen
  and by embedding :ref:`live-data` from our :ref:`labs beehive <labhive-one>` in Berlin Wedding
- Add liability disclaimer for public services
- Add embedded Graphs to :ref:`history` displaying
  :ref:`harvesting <labs-event-20160522-harvesting>` and
  :ref:`nightly weight gain <labs-event-20160523-nightly-weight-gain>` events
- Fix image links to Clemens' Open Hive `Adafruit HUZZAH ESP8266`_
- Add information about the `Open Bee Hive`_ “Bienenkisten-Monitoring”
  project by Markus Euskirchen to :ref:`history`
- Add more information about the `Open Hive`_
  project by Clemens Gruber to :ref:`history`
- Rename referenced Grafana dashboard to “`hiveeyes-labs-wedding <Hiveeyes Labs Wedding Dashboard_>`_”,
  remove “fullscreen” query parameter for Grafana 3.0 compatibility
- Improve “ptrace” make target: Proper filename shell- and url-escaping
- Significant updates to :ref:`hiveeyes-system`,
  :ref:`hiveeyes-one`, :ref:`open-hive` and :ref:`history` pages
- Add :ref:`people` page
- Replace unix times (epoch) by human-readable time format
  (UTC string) in links to Grafana reference dashboard
- Tweak and polish documentation
- Improve coherency and interlinking between resources
- Add important details
- Start and refactor content into new pages
  :ref:`handbook`, :ref:`do-it-yourself`, :ref:`goals`,
  :ref:`development`, :ref:`labhive-one` and :ref:`terms`


2016-05-23 0.3.0
================
- Update :ref:`platform-setup` re. `Hiveeyes Debian package repository`_
- Update project history


2016-03-07 0.2.0
================
- Improve project history
- Add intersphinx connections to BERadio_ and Kotori_
  and start interlinking documentation between associated projects
- Migrate information from splash screen into system docs
- Update splash screen
- Add infos about technology foundation and research notepad
- Update designated InfluxDB version to 0.10.2


2016-02-22 0.1.1
================
- Add bumpversion for convenient version bumping
- Add CHANGES.rst
- Improve main index page and wording


2016-02-22 0.1.0
================
- Initial documentation release
- Aggregate information collected during the last three years
  across different projects.



src/clients/php5/example.php
<?php
// -*- coding: utf-8 -*-
/*
============================
Hiveeyes PHP example program
============================

Documentation
-------------
- https://hiveeyes.org/docs/system/acquisition/php.html#demo

Downloads
---------
- https://hiveeyes.org/docs/system/_downloads/example.php
- https://hiveeyes.org/docs/system/_downloads/hiveeyes.php
- https://getkotori.org/docs/_downloads/terkin-http.php


Synopsis
--------
Run demonstration program from the command line::

    # Send fixed measurements "temperature" => 42.84, "humidity" => 83 for demonstration purposes
    php -f src/clients/php5/example.php run demo

    # Send a periodic, slowly oscillating sawtooth signal
    php -f src/clients/php5/example.php run sawtooth

*/

namespace HiveeyesDemo {

    // Add search path for finding foundation library terkin-http.php
    $path = '/Users/amo/dev/isarengineering/sources/kotori/clients/runtime/php';
    set_include_path(get_include_path() . PATH_SEPARATOR . $path);

    // Put this file into the folder of your PHP program
    include("hiveeyes.php");

    // Acquire API library
    use Hiveeyes\HiveeyesNode;

    // When running from the command line, use some example programs
    // for submitting telemetry data for demonstration purposes.
    if (php_sapi_name() == "cli") {

        $telemetry = new HiveeyesNode(
            array(
                "network"   => "testdrive",
                "gateway"   => "area-42",
                "node"      => "node-1",
            )
        );

        if ($argc > 2) {
            $command = $argv[1];
            $subcommand = $argv[2];
        }

        if ($command == "run") {

            if ($subcommand == "demo") {
                // Emit single sample of a sawtooth signal
                $data = array("temperature" => 42.84, "humidity" => 83);
                var_dump($telemetry->transmit($data));

            } else if ($subcommand == "sawtooth") {
                date_default_timezone_set("Europe/Berlin");
                // Emit sample of a sawtooth signal each second, periodically
                while (true) {
                    $data = array("second" => intval(strftime("%S")));
                    var_dump($telemetry->transmit($data));
                    sleep(1);
                }
            }
        }

    }

}

?>


src/clients/php5/hiveeyes.php
<?php
// -*- coding: utf-8 -*-
/*
======================
Hiveeyes PHP data node
======================

Documentation
-------------
- https://hiveeyes.org/docs/system/acquisition/php.html#library

Downloads
---------
- https://hiveeyes.org/docs/system/_downloads/hiveeyes.php
- https://getkotori.org/docs/_downloads/terkin-http.php

Synopsis
--------
Transmit telemetry data from PHP::

    // Put this file into the folder of your PHP program
    include("hiveeyes.php");

    // Acquire API library
    use Hiveeyes\HiveeyesNode;

    // Create a "Node API" telemetry client object
    $telemetry = new HiveeyesNode(
        array(
            "network"   => "testdrive",
            "gateway"   => "area-42",
            "node"      => "node-1",
        )
    );

    // Transmit data
    $data = array("temperature" => 42.84, "humidity" => 83, "weight" => 37.4);
    $telemetry->transmit($data);


Basic API
---------

    // Create a "Basic API" telemetry client object
    include("terkin-http.php");
    use Terkin\TelemetryClient;
    $telemetry = new TelemetryClient("https://swarm.hiveeyes.org/api/hiveeyes/testdrive/area-42/node-1/data");

*/

namespace Hiveeyes {

    include("terkin-http.php");
    use Terkin\TelemetryNode;

    class HiveeyesNode extends TelemetryNode {
        /***
         *
         * Telemetry node client: Network participant API
         *
        **/

        function __construct($api_uri, $options = array()) {
            if (is_array($api_uri)) {
                $options = $api_uri;
                $api_uri = "https://swarm.hiveeyes.org/api";
            }
            $options['realm'] = 'hiveeyes';
            parent::__construct($api_uri, $options);
        }

    }

}

?>
